app.rb

```diff
@@ -0,0 +1,149 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def start(menu_options, display_menu_method)
+    puts 'Welcome to the School Library App!'
+    loop do
+      display_menu_method.call
+      choice = gets.chomp
+
+      if menu_options.include?(choice)
+        send(menu_options[choice])
+      else
+        puts 'Invalid choice. Please try again.'
+      end
+    end
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'No books available. Please add some books and try again.'
+    else
+      puts 'List of available books:'
+      @books.each_with_index do |book, index|
+        puts "#{index + 1}) Title: #{book.title} | Author: #{book.author}"
+      end
+    end
+  end
+
+  def list_people
+    if @people.empty?
+      puts 'No people added yet. Please add a person and try again.'
+    else
+      puts 'List of people:'
+      @people.each_with_index do |person, index|
+        puts "#{index + 1}) [#{person.class}] Name: #{person.name} | Age: #{person.age} | ID: #{person.id}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Would you like to create a student (1) or a teacher (2)? Select a number: '
+    choice = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    case choice
+    when '1'
+      create_student(name, age)
+    when '2'
+      create_teacher(name, age)
+    else
+      puts 'Invalid choice. Please try again.'
+    end
+  end
+
+  def create_student(name, age)
+    classroom = nil
+    student = Student.new(age, classroom, name)
+    @people << student
+    puts "Student created successfully. ID is #{student.id}"
+  end
+
+  def create_teacher(name, age)
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(age, specialization, name) # Corrected the order of arguments
+    @people << teacher
+    puts "Teacher created successfully. Teacher ID is #{teacher.id}"
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+
+    if title.strip != '' && author.strip != ''
+      book = Book.new(title, author)
+      @books << book
+      puts 'Book created successfully.'
+    else
+      puts 'Please enter the book title and author.'
+    end
+  end
+
+  def create_rental
+    if @books.empty? || @people.empty?
+      puts 'Nothing to see here. Add books and people first.'
+    else
+      puts 'Select a book to rent:'
+      list_books
+      book_number = gets.chomp.to_i - 1
+      puts 'Enter your ID to rent the book:'
```

Comment:
[OPTIONAL]
This is not a major change, but I recommend changing the text so that users don't make mistake by selecting ID.

 - Here you should write to select a person, instead of ID. If I put the ID here then it will not going to work.